dbtest/dbserver.go

```diff
@@ -110,12 +113,27 @@ func (dbs *DBServer) start() {
 		fmt.Fprintf(os.Stderr, "mongod failed to start: %v\n", err)
 		panic(err)
 	}
+	// Give the db time to settle.  This seems to matter on docker instances.
+	time.Sleep(1 * time.Second)
+	if repl {
+		dbs.initiateRepl(addr.Port)
+	}
 	if !dbs.disableMonitor {
 		dbs.tomb.Go(dbs.monitor)
 	}
 	dbs.Wipe()
 }
 
+func (dbs *DBServer) initiateRepl(port int) {
+	args := []string{
+		"localhost:" + strconv.Itoa(port),
+		"--eval", "rs.initiate()",
+	}
+	shell := exec.Command("mongo", args...)
```

Comment:
Note that instead of shelling out to 'mongo' here, you can do:

func (dbs *DBServer) initiateRepl() {
	session, err := mgo.DialWithInfo(&mgo.DialInfo{
		Addrs:    []string{dbs.host},
		Direct:   true, // must do Direct=true when dealing with a replicaset
		Timeout:  10 * time.Second,
		Database: "test",
		// We don't set ReplicaSetName here, because the replicaset hasn't actually
		// been initialized yet.
		// ReplicaSetName: dbs.replset,
	})
	if err == nil {
		defer session.Close()
		session.SetMode(mgo.Monotonic, true)
		err := session.Run(bson.D{{Name: "replSetInitiate", Value: nil}}, nil)
		if err != nil {
			panic(err)
		}
	} else {
		panic(err)
	}
}

That is functionally equivalent, but means we are using the mgo driver to connect instead of having a separate executable. I realize the test suite already uses 'mongo ... ' to evaluate the wait/dropall etc